File: src/main/java/org/occidere/githubnotifier/vo/GithubFollower.java
package org.occidere.githubnotifier.vo;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author occidere
 * @Blog: https://blog.naver.com/occidere
 * @Github: https://github.com/occidere
 * @since 2019. 12. 04.
 */
@Getter
@Setter
@ToString
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class GithubFollower {

    private String login;
    private long id;
    @JsonProperty("node_id")
    private String nodeId;
    @JsonProperty("avatar_url")
    private String avatarUrl;
    @JsonProperty("html_url")
    private String htmlUrl;
    private String type;
} // FIXME: Find why are not indexed @JsonProperty fields

File: src/main/scala/org/occidere/githubnotifier/vo/GithubFollower.scala
package org.occidere.githubnotifier.vo

import com.fasterxml.jackson.annotation.{JsonIgnoreProperties, JsonInclude, JsonProperty}
import lombok.ToString

/**
 * @author occidere
 * @Blog: https://blog.naver.com/occidere
 * @Github: https://github.com/occidere
 * @since 2020. 08. 15.
 */
@ToString
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
class GithubFollower {
  val login: String = null
  val id: Long = -1
  @JsonProperty("node_id")
  val nodeId: String = null
  @JsonProperty("avatar_url")
  val avatarUrl: String = null
  @JsonProperty("html_url")
  val htmlUrl: String = null
  val `type`: String = null // FIXME
}
